packages/actor-http-cache/lib/ActorHttpCache.ts

```diff
@@ -0,0 +1,205 @@
+/* eslint import/no-nodejs-modules: ["error", {"allow": ["stream"]}] */
+import { Readable } from 'stream';
+import type {
+  IActionHttp,
+  IActorHttpOutput,
+  IActorHttpArgs,
+} from '@comunica/bus-http';
+import { ActorHttp } from '@comunica/bus-http';
+import type {
+  ActorHttpInvalidateListenable,
+  IActionHttpInvalidate,
+  MediatorHttpInvalidate,
+} from '@comunica/bus-http-invalidate';
+import type { IHttpCacheStorage } from '@comunica/http-cache-storage';
+import type { IMediatorTypeTime } from '@comunica/mediatortype-time';
+import 'cross-fetch/polyfill';
+import * as CachePolicy from 'http-cache-semantics';
+import {
+  requestToRequestWithHashHeaders,
+  responseToResponseWithHashHeaders,
+  addHashHeadersToObject,
+} from './headerConversionHelpers';
+
+/**
+ * A comunica Cache Http Actor.
+ */
+export class ActorHttpCache extends ActorHttp {
+  private readonly cacheStorage: IHttpCacheStorage;
+  private readonly mediatorHttpInvalidate: MediatorHttpInvalidate;
+  private readonly actorHttpFetch: ActorHttp;
+
+  public constructor(args: IActorHttpCacheArgs) {
+    super(args);
+    this.cacheStorage = args.cacheStorage;
```

Comment:
As described in #166, we should also cover the case where a custom `cacheStorage` is supplied by the user inside the context object (in the `run` method).

(feel free to ask me for pointers if this is not fully clear)

Reply:
Did something change from the conversation here https://github.com/comunica/comunica/issues/166#issuecomment-1151448769? I'm happy to implement a custom cache storage, but at the moment, I don't see how it would work.

Reply:
> Did something change from the conversation here https://github.com/comunica/comunica/issues/166#issuecomment-1151448769?

No, that still holds.

I did not mean that we should implement a custom cache, but that it should be possible for users to pass a custom cache (that implements the new `IHttpCacheStorage` interface) via the context. In the same way users can provide options such as baseIri and so on.

Reply:
Got it. Updated